Tensile/TensileCreateLibrary.py

```diff
@@ -1320,46 +1361,30 @@ def splitExtraParameters(par):
       print1(f"codeObjectFiles: {codeObjectFiles}")
       print1(f"sourceLibPaths + asmLibPaths: {sourceLibPaths + asmLibPaths}")
 
+  # do we need this or have we already done this?
   archs = [gfxName(arch) for arch in globalParameters['SupportedISA'] \
              if globalParameters["AsmCaps"][arch]["SupportedISA"]]
-  newLibraryDir = ensurePath(os.path.join(outputPath, 'library'))
 
-  if globalParameters["SeparateArchitectures"] or globalParameters["LazyLibraryLoading"]:
-    for archName, newMasterLibrary in masterLibraries.items():
-      if archName in archs:
-        if globalParameters["LazyLibraryLoading"]:
-          masterFile = os.path.join(newLibraryDir, "TensileLibrary_lazy_"+archName)
-        else:
-          masterFile = os.path.join(newLibraryDir, "TensileLibrary_"+archName)
-        newMasterLibrary.applyNaming(kernelMinNaming)
-        LibraryIO.write(masterFile, Utils.state(newMasterLibrary), args.LibraryFormat)
-
-        #Write placeholder libraries
-        for name, lib in newMasterLibrary.lazyLibraries.items():
-          filename = os.path.join(newLibraryDir, name)
-          lib.applyNaming(kernelMinNaming) #@TODO Check to see if kernelMinNaming is correct
-          LibraryIO.write(filename, Utils.state(lib), args.LibraryFormat)
-
-  else:
-    masterFile = os.path.join(newLibraryDir, "TensileLibrary")
-    fullMasterLibrary.applyNaming(kernelMinNaming)
-    LibraryIO.write(masterFile, Utils.state(fullMasterLibrary), args.LibraryFormat)
+  newLibraryDir = Path(outputPath) / "library"
+  newLibraryDir.mkdir(exist_ok=True)
 
-  theMasterLibrary = fullMasterLibrary
-  if globalParameters["SeparateArchitectures"]:
-    theMasterLibrary = list(masterLibraries.values())[0]
+  masterFileList = generateMasterFileList(masterLibraries, archs, args.LazyLibraryLoading) if args.SeparateArchitectures \
+    else [("TensileLibrary", fullMasterLibrary)]
+  for name, lib in masterFileList:
+    writeMasterFile(newLibraryDir, args.LibraryFormat, kernelMinNaming, name, lib)
+  masterFile, fullMasterLibrary = masterFileList[0]
```

Comment:
Has the consistency with previous implementation been tested for the various combinations of separate architectures, lazy loading, etc. to ensure the output is unchanged when using this later in embed library?